src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -32,21 +32,24 @@ class EntityUserProvider implements UserProviderInterface
     private $classOrAlias;
     private $class;
     private $property;
+    private $repository;
 
-    public function __construct(ManagerRegistry $registry, $classOrAlias, $property = null, $managerName = null)
+    public function __construct(ManagerRegistry $registry, $classOrAlias, $property = null, $managerName = null, $repository = null)
```

Comment:
can it be typehinted, e.g. `EntityRepository` or so

Reply:
This PR enables using custom entity repositories that don't extend `EntityRepository`, so the parameter can't be typehinted

Reply:
im not sure.. a custom doctrine repository should still extend from EntityRepository AFAIK. Here we need to be sure we can call `$repository->find()` and `$repository->findOneBy()`. No?